modules/post/linux/gather/busybox_pingnet.rb

```diff
@@ -0,0 +1,165 @@
+##
+# This module requires Metasploit: http://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'msf/core'
+
+class Metasploit3 < Msf::Post
+
+  include Msf::Post::File
+
+  def initialize
+    super(
+      'Name'         => 'BusyBox Ping Network',
+      'Description'  => 'This module will be applied on a session connected
+                         to a BusyBox sh shell. The script will ping a range of
+                         ip adresses from the router or device executing BusyBox.',
+      'Author'       => 'Javier Vicente Vallejo',
+      'License'      => MSF_LICENSE,
+      'References'   =>
+        [
+          [ 'URL', 'http://vallejo.cc']
+        ],
+      'Platform'      => ['linux'],
+       'SessionTypes'  => ['shell']
+    )
+
+    register_options(
+      [
+        OptAddress.new('IPRANGESTART',   [ true, "The first ip address of the range to ping.", nil ]),
+        OptAddress.new('IPRANGEEND',   [ true, "The last ip address of the range to ping.", nil ])
+      ], self.class)
+
+  end
+
+  def run
+
+    #this module will send a sh script for busybox shell for doing ping to a range of ip address from
+    #the router or device that is executing busybox. It could be possible to calculate each ip address
+    #of the range of ip addresses in the ruby script and execute each ping command with cmd_exec, but
+    #it would generate an unnecesary traffic in the connection with the busybox device (usually telnet)
```

Comment:
And using session.shell_write and session. shell_read won't generate traffic :? I guess I'm forgetting something, so just asking.